staker/stateless_block_validator.go

```diff
@@ -153,15 +154,15 @@ func (e *validationEntry) ToInput(stylusArchs []string) (*validator.ValidationIn
 	for _, stylusArch := range stylusArchs {
 		res.UserWasms[stylusArch] = make(map[common.Hash][]byte)
 	}
-	for hash, info := range e.UserWasms {
+	localTarget := programs.LocalTargetName()
+	for hash, asmMap := range e.UserWasms {
 		for _, stylusArch := range stylusArchs {
-			if stylusArch == "wavm" {
-				res.UserWasms[stylusArch][hash] = info.Module
-			} else if stylusArch == runtime.GOARCH {
-				res.UserWasms[stylusArch][hash] = info.Asm
-			} else {
+			if stylusArch != rawdb.TargetWavm && stylusArch != localTarget {
```

Comment:
this one shouldn't actually compare to localTarget, as stylusArchs could come from a remote validator that could be of a different architecture.
Just make sure that all requested stylusArchs exist, update them in the input and return an error if they don't